hack/docs/configdoc.go

```diff
@@ -0,0 +1,392 @@
+	"strings"
+)
+
+// Set this to the root of the repo
+const configDir = "config"
+
+//go:embed workflow-controller-configmap.md
+var header string
+
+const outputFile = "docs/workflow-controller-configmap.md"
+
+// visited tracks which types we've already documented
+var visited = map[string]bool{}
+
+// typeSpecs maps type name to its *ast.TypeSpec
+var typeSpecs = map[string]*ast.TypeSpec{}
+
+// typeComments maps type name to its documentation comment
+var typeComments = map[string]*ast.CommentGroup{}
+
+// documentedTypes tracks which types we will document in this file
+var documentedTypes = map[string]bool{}
+
+func generateConfigDocs() {
+	fset := token.NewFileSet()
+
+	// Parse all .go files in the config directory
+	err := filepath.Walk(configDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		// Skip test files and non-go files
+		if !strings.HasSuffix(path, ".go") || strings.HasSuffix(path, "_test.go") {
+			return nil
+		}
+
+		f, err := parser.ParseFile(fset, path, nil, parser.ParseComments)
+		if err != nil {
+			log.Printf("failed to parse %s: %v", path, err)
+			return nil // Continue with other files
+		}
+
+		// Collect all type specs in this file
+		for _, decl := range f.Decls {
+			gd, ok := decl.(*ast.GenDecl)
+			if !ok {
+				continue
+			}
+			for _, spec := range gd.Specs {
+				ts, ok := spec.(*ast.TypeSpec)
+				if ok {
+					typeSpecs[ts.Name.Name] = ts
+					// Associate the GenDecl's comment with this type
+					if gd.Doc != nil {
+						typeComments[ts.Name.Name] = gd.Doc
+					}
+					documentedTypes[ts.Name.Name] = true
+				}
+			}
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		log.Fatalf("failed to walk config directory: %v", err)
+	}
+
+	var buf bytes.Buffer
+	buf.WriteString(header)
+	if ts, ok := typeSpecs["Config"]; ok {
+		writeStructDoc(&buf, ts, "Config")
+	} else {
+		log.Fatalf("Config struct not found in %s directory", configDir)
+	}
+
+	err = os.WriteFile(outputFile, buf.Bytes(), 0644)
+	if err != nil {
+		log.Fatalf("failed to write output: %v", err)
+	}
+	fmt.Printf("Wrote %s\n", outputFile)
+}
+
+func writeStructDoc(buf *bytes.Buffer, ts *ast.TypeSpec, name string) {
+	if visited[name] {
+		return
+	}
+	visited[name] = true
+
+	// Check if this is actually a struct type first
+	st, ok := ts.Type.(*ast.StructType)
+	if !ok {
+		// This is not a struct (e.g., type alias like TTL or MetricsTemporality)
+		// Don't create a section for it
+		return
+	}
+
+	buf.WriteString(fmt.Sprintf("\n## %s\n\n", name))
+	// Check for comment from GenDecl first, then TypeSpec
+	if comment, ok := typeComments[name]; ok && comment != nil {
+		buf.WriteString(normalizeComment(comment.Text()) + "\n\n")
+	} else if ts.Doc != nil {
+		buf.WriteString(normalizeComment(ts.Doc.Text()) + "\n\n")
+	}
+	buf.WriteString("### Fields\n\n")
+	buf.WriteString("| Field Name | Field Type | Description |\n")
+	buf.WriteString("|:----------:|:----------:|:------------|\n")
```

Comment:
Fine I suppose, but I thought we maybe had some library to better manipulate markdown?

Reply:
Good point. I remember checking that library and rejecting it, but looking again now it feels right. I'll update

Reply:
Done